gamemodes/terrortown/gamemode/server/sv_main.lua

```diff
@@ -840,38 +840,40 @@ function TellTraitorsAboutTraitors()
 			traitornicks[#traitornicks + 1] = v:Nick()
 		end
 	end
+	
+	for _, v in ipairs(player.GetAll()) do
+		if v:HasTeam(TEAM_TRAITOR) then
 
-	hook.Run("TTT2TellTraitors", traitornicks)
+			local tmp, shouldShow = hook.Run("TTT2TellTraitors", v, table.Copy(traitornicks))
 
-	-- This is ugly as hell, but it's kinda nice to filter out the names of the
-	-- traitors themselves in the messages to them
-	local traitornicks_min = #traitornicks < 2
+			if shouldShow == nil then shouldShow = true end
+			if not shouldShow then continue end
 
-	if not traitornicks_min then
-		table.sort(traitornicks, function(a, b)
-			return a and b and a:upper() < b:upper()
-		end)
-	end
+			if tmp == nil then tmp = table.Copy(traitornicks) end
 
-	for _, v in ipairs(player.GetAll()) do
-		if v:HasTeam(TEAM_TRAITOR) then
-			if traitornicks_min then
-				LANG.Msg(v, "round_traitors_one")
+			table.RemoveByValue(tmp, v:Nick())
```

Comment:
I never used this function :D

Reply:
Would be nice to handle it in another way as I did, this functions is looping through the table. This means a duplicated amount of processes